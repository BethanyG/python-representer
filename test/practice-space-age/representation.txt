Module([FunctionDef('placeholder_0', arguments([], [arg('placeholder_1', lineno=1, col_offset=21, end_lineno=1, end_col_offset=27)], kwonlyargs=[], kw_defaults=[], defaults=[]), [FunctionDef('placeholder_2', arguments([], [arg('self', lineno=2, col_offset=14, end_lineno=2, end_col_offset=18)], kwonlyargs=[], kw_defaults=[], defaults=[]), [Return(Call(Name('round', Load(), lineno=3, col_offset=15, end_lineno=3, end_col_offset=20), [BinOp(Attribute(Name('self', Load(), lineno=3, col_offset=21, end_lineno=3, end_col_offset=25), 'seconds', Load(), lineno=3, col_offset=21, end_lineno=3, end_col_offset=33), Div(), Name('placeholder_1', Load(), lineno=3, col_offset=36, end_lineno=3, end_col_offset=42), lineno=3, col_offset=21, end_lineno=3, end_col_offset=42), Constant(2, lineno=3, col_offset=44, end_lineno=3, end_col_offset=45)], [], lineno=3, col_offset=15, end_lineno=3, end_col_offset=46), lineno=3, col_offset=8, end_lineno=3, end_col_offset=46)], [], lineno=2, col_offset=4, end_lineno=3, end_col_offset=46), Return(Name('placeholder_2', Load(), lineno=5, col_offset=11, end_lineno=5, end_col_offset=16), lineno=5, col_offset=4, end_lineno=5, end_col_offset=16)], [], lineno=1, col_offset=0, end_lineno=5, end_col_offset=16), ClassDef('placeholder_3', [], [], [Assign([Name('placeholder_4', Store(), lineno=10, col_offset=4, end_lineno=10, end_col_offset=14)], Call(Name('placeholder_0', Load(), lineno=10, col_offset=17, end_lineno=10, end_col_offset=33), [Constant(7600530.24, lineno=10, col_offset=34, end_lineno=10, end_col_offset=44)], [], lineno=10, col_offset=17, end_lineno=10, end_col_offset=45), lineno=10, col_offset=4, end_lineno=10, end_col_offset=45), Assign([Name('placeholder_5', Store(), lineno=11, col_offset=4, end_lineno=11, end_col_offset=12)], Call(Name('placeholder_0', Load(), lineno=11, col_offset=15, end_lineno=11, end_col_offset=31), [Constant(19413907.2, lineno=11, col_offset=32, end_lineno=11, end_col_offset=42)], [], lineno=11, col_offset=15, end_lineno=11, end_col_offset=43), lineno=11, col_offset=4, end_lineno=11, end_col_offset=43), Assign([Name('placeholder_6', Store(), lineno=12, col_offset=4, end_lineno=12, end_col_offset=12)], Call(Name('placeholder_0', Load(), lineno=12, col_offset=15, end_lineno=12, end_col_offset=31), [Constant(31558149.76, lineno=12, col_offset=32, end_lineno=12, end_col_offset=43)], [], lineno=12, col_offset=15, end_lineno=12, end_col_offset=44), lineno=12, col_offset=4, end_lineno=12, end_col_offset=44), Assign([Name('placeholder_7', Store(), lineno=13, col_offset=4, end_lineno=13, end_col_offset=11)], Call(Name('placeholder_0', Load(), lineno=13, col_offset=14, end_lineno=13, end_col_offset=30), [Constant(59354294.4, lineno=13, col_offset=31, end_lineno=13, end_col_offset=41)], [], lineno=13, col_offset=14, end_lineno=13, end_col_offset=42), lineno=13, col_offset=4, end_lineno=13, end_col_offset=42), Assign([Name('placeholder_8', Store(), lineno=14, col_offset=4, end_lineno=14, end_col_offset=14)], Call(Name('placeholder_0', Load(), lineno=14, col_offset=17, end_lineno=14, end_col_offset=33), [Constant(374335776.0, lineno=14, col_offset=34, end_lineno=14, end_col_offset=45)], [], lineno=14, col_offset=17, end_lineno=14, end_col_offset=46), lineno=14, col_offset=4, end_lineno=14, end_col_offset=46), Assign([Name('placeholder_9', Store(), lineno=15, col_offset=4, end_lineno=15, end_col_offset=13)], Call(Name('placeholder_0', Load(), lineno=15, col_offset=16, end_lineno=15, end_col_offset=32), [Constant(929596608.0, lineno=15, col_offset=33, end_lineno=15, end_col_offset=44)], [], lineno=15, col_offset=16, end_lineno=15, end_col_offset=45), lineno=15, col_offset=4, end_lineno=15, end_col_offset=45), Assign([Name('placeholder_10', Store(), lineno=16, col_offset=4, end_lineno=16, end_col_offset=13)], Call(Name('placeholder_0', Load(), lineno=16, col_offset=16, end_lineno=16, end_col_offset=32), [Constant(2651370019.3, lineno=16, col_offset=33, end_lineno=16, end_col_offset=45)], [], lineno=16, col_offset=16, end_lineno=16, end_col_offset=46), lineno=16, col_offset=4, end_lineno=16, end_col_offset=46), Assign([Name('placeholder_11', Store(), lineno=17, col_offset=4, end_lineno=17, end_col_offset=14)], Call(Name('placeholder_0', Load(), lineno=17, col_offset=17, end_lineno=17, end_col_offset=33), [Constant(5200418592.0, lineno=17, col_offset=34, end_lineno=17, end_col_offset=46)], [], lineno=17, col_offset=17, end_lineno=17, end_col_offset=47), lineno=17, col_offset=4, end_lineno=17, end_col_offset=47), FunctionDef('placeholder_12', arguments([], [arg('self', lineno=19, col_offset=17, end_lineno=19, end_col_offset=21), arg('placeholder_13', lineno=19, col_offset=23, end_lineno=19, end_col_offset=30)], kwonlyargs=[], kw_defaults=[], defaults=[]), [Assign([Attribute(Name('self', Load(), lineno=20, col_offset=8, end_lineno=20, end_col_offset=12), 'placeholder_13', Store(), lineno=20, col_offset=8, end_lineno=20, end_col_offset=20)], Name('placeholder_13', Load(), lineno=20, col_offset=23, end_lineno=20, end_col_offset=30), lineno=20, col_offset=8, end_lineno=20, end_col_offset=30)], [], lineno=19, col_offset=4, end_lineno=20, end_col_offset=30)], [], lineno=8, col_offset=0, end_lineno=20, end_col_offset=30)], [])