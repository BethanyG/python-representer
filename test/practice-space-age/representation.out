## BEGIN NORMALIZED CODE ##
def placeholder_0(placeholder_1):
    def placeholder_2(self):
        return round(self.seconds / placeholder_1, 2)

    return placeholder_2


class placeholder_3:
    placeholder_4 = placeholder_0(7600530.24)
    placeholder_5 = placeholder_0(19413907.2)
    placeholder_6 = placeholder_0(31558149.76)
    placeholder_7 = placeholder_0(59354294.4)
    placeholder_8 = placeholder_0(374335776.0)
    placeholder_9 = placeholder_0(929596608.0)
    placeholder_10 = placeholder_0(2651370019.3)
    placeholder_11 = placeholder_0(5200418592.0)

    def placeholder_12(self, placeholder_13):
        self.placeholder_13 = placeholder_13

## END NORMALIZED CODE ##

## BEGIN TREE BEFORE ##
Module(
  [
    FunctionDef(
      'period_converter',
      arguments(
        [],
        [
          arg(
            'period',
            lineno=1,
            col_offset=21,
            end_lineno=1,
            end_col_offset=27)],
        kwonlyargs=[],
        kw_defaults=[],
        defaults=[]),
      [
        FunctionDef(
          'inner',
          arguments(
            [],
            [
              arg(
                'self',
                lineno=2,
                col_offset=14,
                end_lineno=2,
                end_col_offset=18)],
            kwonlyargs=[],
            kw_defaults=[],
            defaults=[]),
          [
            Return(
              Call(
                Name(
                  'round',
                  Load(),
                  lineno=3,
                  col_offset=15,
                  end_lineno=3,
                  end_col_offset=20),
                [
                  BinOp(
                    Attribute(
                      Name(
                        'self',
                        Load(),
                        lineno=3,
                        col_offset=21,
                        end_lineno=3,
                        end_col_offset=25),
                      'seconds',
                      Load(),
                      lineno=3,
                      col_offset=21,
                      end_lineno=3,
                      end_col_offset=33),
                    Div(),
                    Name(
                      'period',
                      Load(),
                      lineno=3,
                      col_offset=36,
                      end_lineno=3,
                      end_col_offset=42),
                    lineno=3,
                    col_offset=21,
                    end_lineno=3,
                    end_col_offset=42),
                  Constant(
                    2,
                    lineno=3,
                    col_offset=44,
                    end_lineno=3,
                    end_col_offset=45)],
                [],
                lineno=3,
                col_offset=15,
                end_lineno=3,
                end_col_offset=46),
              lineno=3,
              col_offset=8,
              end_lineno=3,
              end_col_offset=46)],
          [],
          lineno=2,
          col_offset=4,
          end_lineno=3,
          end_col_offset=46),
        Return(
          Name(
            'inner',
            Load(),
            lineno=5,
            col_offset=11,
            end_lineno=5,
            end_col_offset=16),
          lineno=5,
          col_offset=4,
          end_lineno=5,
          end_col_offset=16)],
      [],
      lineno=1,
      col_offset=0,
      end_lineno=5,
      end_col_offset=16),
    ClassDef(
      'SpaceAge',
      [],
      [],
      [
        Assign(
          [
            Name(
              'on_mercury',
              Store(),
              lineno=10,
              col_offset=4,
              end_lineno=10,
              end_col_offset=14)],
          Call(
            Name(
              'period_converter',
              Load(),
              lineno=10,
              col_offset=17,
              end_lineno=10,
              end_col_offset=33),
            [
              Constant(
                7600530.24,
                lineno=10,
                col_offset=34,
                end_lineno=10,
                end_col_offset=44)],
            [],
            lineno=10,
            col_offset=17,
            end_lineno=10,
            end_col_offset=45),
          lineno=10,
          col_offset=4,
          end_lineno=10,
          end_col_offset=45),
        Assign(
          [
            Name(
              'on_venus',
              Store(),
              lineno=11,
              col_offset=4,
              end_lineno=11,
              end_col_offset=12)],
          Call(
            Name(
              'period_converter',
              Load(),
              lineno=11,
              col_offset=15,
              end_lineno=11,
              end_col_offset=31),
            [
              Constant(
                19413907.2,
                lineno=11,
                col_offset=32,
                end_lineno=11,
                end_col_offset=42)],
            [],
            lineno=11,
            col_offset=15,
            end_lineno=11,
            end_col_offset=43),
          lineno=11,
          col_offset=4,
          end_lineno=11,
          end_col_offset=43),
        Assign(
          [
            Name(
              'on_earth',
              Store(),
              lineno=12,
              col_offset=4,
              end_lineno=12,
              end_col_offset=12)],
          Call(
            Name(
              'period_converter',
              Load(),
              lineno=12,
              col_offset=15,
              end_lineno=12,
              end_col_offset=31),
            [
              Constant(
                31558149.76,
                lineno=12,
                col_offset=32,
                end_lineno=12,
                end_col_offset=43)],
            [],
            lineno=12,
            col_offset=15,
            end_lineno=12,
            end_col_offset=44),
          lineno=12,
          col_offset=4,
          end_lineno=12,
          end_col_offset=44),
        Assign(
          [
            Name(
              'on_mars',
              Store(),
              lineno=13,
              col_offset=4,
              end_lineno=13,
              end_col_offset=11)],
          Call(
            Name(
              'period_converter',
              Load(),
              lineno=13,
              col_offset=14,
              end_lineno=13,
              end_col_offset=30),
            [
              Constant(
                59354294.4,
                lineno=13,
                col_offset=31,
                end_lineno=13,
                end_col_offset=41)],
            [],
            lineno=13,
            col_offset=14,
            end_lineno=13,
            end_col_offset=42),
          lineno=13,
          col_offset=4,
          end_lineno=13,
          end_col_offset=42),
        Assign(
          [
            Name(
              'on_jupiter',
              Store(),
              lineno=14,
              col_offset=4,
              end_lineno=14,
              end_col_offset=14)],
          Call(
            Name(
              'period_converter',
              Load(),
              lineno=14,
              col_offset=17,
              end_lineno=14,
              end_col_offset=33),
            [
              Constant(
                374335776.0,
                lineno=14,
                col_offset=34,
                end_lineno=14,
                end_col_offset=45)],
            [],
            lineno=14,
            col_offset=17,
            end_lineno=14,
            end_col_offset=46),
          lineno=14,
          col_offset=4,
          end_lineno=14,
          end_col_offset=46),
        Assign(
          [
            Name(
              'on_saturn',
              Store(),
              lineno=15,
              col_offset=4,
              end_lineno=15,
              end_col_offset=13)],
          Call(
            Name(
              'period_converter',
              Load(),
              lineno=15,
              col_offset=16,
              end_lineno=15,
              end_col_offset=32),
            [
              Constant(
                929596608.0,
                lineno=15,
                col_offset=33,
                end_lineno=15,
                end_col_offset=44)],
            [],
            lineno=15,
            col_offset=16,
            end_lineno=15,
            end_col_offset=45),
          lineno=15,
          col_offset=4,
          end_lineno=15,
          end_col_offset=45),
        Assign(
          [
            Name(
              'on_uranus',
              Store(),
              lineno=16,
              col_offset=4,
              end_lineno=16,
              end_col_offset=13)],
          Call(
            Name(
              'period_converter',
              Load(),
              lineno=16,
              col_offset=16,
              end_lineno=16,
              end_col_offset=32),
            [
              Constant(
                2651370019.3,
                lineno=16,
                col_offset=33,
                end_lineno=16,
                end_col_offset=45)],
            [],
            lineno=16,
            col_offset=16,
            end_lineno=16,
            end_col_offset=46),
          lineno=16,
          col_offset=4,
          end_lineno=16,
          end_col_offset=46),
        Assign(
          [
            Name(
              'on_neptune',
              Store(),
              lineno=17,
              col_offset=4,
              end_lineno=17,
              end_col_offset=14)],
          Call(
            Name(
              'period_converter',
              Load(),
              lineno=17,
              col_offset=17,
              end_lineno=17,
              end_col_offset=33),
            [
              Constant(
                5200418592.0,
                lineno=17,
                col_offset=34,
                end_lineno=17,
                end_col_offset=46)],
            [],
            lineno=17,
            col_offset=17,
            end_lineno=17,
            end_col_offset=47),
          lineno=17,
          col_offset=4,
          end_lineno=17,
          end_col_offset=47),
        FunctionDef(
          '__init__',
          arguments(
            [],
            [
              arg(
                'self',
                lineno=19,
                col_offset=17,
                end_lineno=19,
                end_col_offset=21),
              arg(
                'seconds',
                lineno=19,
                col_offset=23,
                end_lineno=19,
                end_col_offset=30)],
            kwonlyargs=[],
            kw_defaults=[],
            defaults=[]),
          [
            Assign(
              [
                Attribute(
                  Name(
                    'self',
                    Load(),
                    lineno=20,
                    col_offset=8,
                    end_lineno=20,
                    end_col_offset=12),
                  'seconds',
                  Store(),
                  lineno=20,
                  col_offset=8,
                  end_lineno=20,
                  end_col_offset=20)],
              Name(
                'seconds',
                Load(),
                lineno=20,
                col_offset=23,
                end_lineno=20,
                end_col_offset=30),
              lineno=20,
              col_offset=8,
              end_lineno=20,
              end_col_offset=30)],
          [],
          lineno=19,
          col_offset=4,
          end_lineno=20,
          end_col_offset=30)],
      [],
      lineno=8,
      col_offset=0,
      end_lineno=20,
      end_col_offset=30)],
  [])
## END TREE BEFORE ##

## BEGIN NORMALIZED TREE ##
Module(
  [
    FunctionDef(
      'placeholder_0',
      arguments(
        [],
        [
          arg(
            'placeholder_1',
            lineno=1,
            col_offset=21,
            end_lineno=1,
            end_col_offset=27)],
        kwonlyargs=[],
        kw_defaults=[],
        defaults=[]),
      [
        FunctionDef(
          'placeholder_2',
          arguments(
            [],
            [
              arg(
                'self',
                lineno=2,
                col_offset=14,
                end_lineno=2,
                end_col_offset=18)],
            kwonlyargs=[],
            kw_defaults=[],
            defaults=[]),
          [
            Return(
              Call(
                Name(
                  'round',
                  Load(),
                  lineno=3,
                  col_offset=15,
                  end_lineno=3,
                  end_col_offset=20),
                [
                  BinOp(
                    Attribute(
                      Name(
                        'self',
                        Load(),
                        lineno=3,
                        col_offset=21,
                        end_lineno=3,
                        end_col_offset=25),
                      'seconds',
                      Load(),
                      lineno=3,
                      col_offset=21,
                      end_lineno=3,
                      end_col_offset=33),
                    Div(),
                    Name(
                      'placeholder_1',
                      Load(),
                      lineno=3,
                      col_offset=36,
                      end_lineno=3,
                      end_col_offset=42),
                    lineno=3,
                    col_offset=21,
                    end_lineno=3,
                    end_col_offset=42),
                  Constant(
                    2,
                    lineno=3,
                    col_offset=44,
                    end_lineno=3,
                    end_col_offset=45)],
                [],
                lineno=3,
                col_offset=15,
                end_lineno=3,
                end_col_offset=46),
              lineno=3,
              col_offset=8,
              end_lineno=3,
              end_col_offset=46)],
          [],
          lineno=2,
          col_offset=4,
          end_lineno=3,
          end_col_offset=46),
        Return(
          Name(
            'placeholder_2',
            Load(),
            lineno=5,
            col_offset=11,
            end_lineno=5,
            end_col_offset=16),
          lineno=5,
          col_offset=4,
          end_lineno=5,
          end_col_offset=16)],
      [],
      lineno=1,
      col_offset=0,
      end_lineno=5,
      end_col_offset=16),
    ClassDef(
      'placeholder_3',
      [],
      [],
      [
        Assign(
          [
            Name(
              'placeholder_4',
              Store(),
              lineno=10,
              col_offset=4,
              end_lineno=10,
              end_col_offset=14)],
          Call(
            Name(
              'placeholder_0',
              Load(),
              lineno=10,
              col_offset=17,
              end_lineno=10,
              end_col_offset=33),
            [
              Constant(
                7600530.24,
                lineno=10,
                col_offset=34,
                end_lineno=10,
                end_col_offset=44)],
            [],
            lineno=10,
            col_offset=17,
            end_lineno=10,
            end_col_offset=45),
          lineno=10,
          col_offset=4,
          end_lineno=10,
          end_col_offset=45),
        Assign(
          [
            Name(
              'placeholder_5',
              Store(),
              lineno=11,
              col_offset=4,
              end_lineno=11,
              end_col_offset=12)],
          Call(
            Name(
              'placeholder_0',
              Load(),
              lineno=11,
              col_offset=15,
              end_lineno=11,
              end_col_offset=31),
            [
              Constant(
                19413907.2,
                lineno=11,
                col_offset=32,
                end_lineno=11,
                end_col_offset=42)],
            [],
            lineno=11,
            col_offset=15,
            end_lineno=11,
            end_col_offset=43),
          lineno=11,
          col_offset=4,
          end_lineno=11,
          end_col_offset=43),
        Assign(
          [
            Name(
              'placeholder_6',
              Store(),
              lineno=12,
              col_offset=4,
              end_lineno=12,
              end_col_offset=12)],
          Call(
            Name(
              'placeholder_0',
              Load(),
              lineno=12,
              col_offset=15,
              end_lineno=12,
              end_col_offset=31),
            [
              Constant(
                31558149.76,
                lineno=12,
                col_offset=32,
                end_lineno=12,
                end_col_offset=43)],
            [],
            lineno=12,
            col_offset=15,
            end_lineno=12,
            end_col_offset=44),
          lineno=12,
          col_offset=4,
          end_lineno=12,
          end_col_offset=44),
        Assign(
          [
            Name(
              'placeholder_7',
              Store(),
              lineno=13,
              col_offset=4,
              end_lineno=13,
              end_col_offset=11)],
          Call(
            Name(
              'placeholder_0',
              Load(),
              lineno=13,
              col_offset=14,
              end_lineno=13,
              end_col_offset=30),
            [
              Constant(
                59354294.4,
                lineno=13,
                col_offset=31,
                end_lineno=13,
                end_col_offset=41)],
            [],
            lineno=13,
            col_offset=14,
            end_lineno=13,
            end_col_offset=42),
          lineno=13,
          col_offset=4,
          end_lineno=13,
          end_col_offset=42),
        Assign(
          [
            Name(
              'placeholder_8',
              Store(),
              lineno=14,
              col_offset=4,
              end_lineno=14,
              end_col_offset=14)],
          Call(
            Name(
              'placeholder_0',
              Load(),
              lineno=14,
              col_offset=17,
              end_lineno=14,
              end_col_offset=33),
            [
              Constant(
                374335776.0,
                lineno=14,
                col_offset=34,
                end_lineno=14,
                end_col_offset=45)],
            [],
            lineno=14,
            col_offset=17,
            end_lineno=14,
            end_col_offset=46),
          lineno=14,
          col_offset=4,
          end_lineno=14,
          end_col_offset=46),
        Assign(
          [
            Name(
              'placeholder_9',
              Store(),
              lineno=15,
              col_offset=4,
              end_lineno=15,
              end_col_offset=13)],
          Call(
            Name(
              'placeholder_0',
              Load(),
              lineno=15,
              col_offset=16,
              end_lineno=15,
              end_col_offset=32),
            [
              Constant(
                929596608.0,
                lineno=15,
                col_offset=33,
                end_lineno=15,
                end_col_offset=44)],
            [],
            lineno=15,
            col_offset=16,
            end_lineno=15,
            end_col_offset=45),
          lineno=15,
          col_offset=4,
          end_lineno=15,
          end_col_offset=45),
        Assign(
          [
            Name(
              'placeholder_10',
              Store(),
              lineno=16,
              col_offset=4,
              end_lineno=16,
              end_col_offset=13)],
          Call(
            Name(
              'placeholder_0',
              Load(),
              lineno=16,
              col_offset=16,
              end_lineno=16,
              end_col_offset=32),
            [
              Constant(
                2651370019.3,
                lineno=16,
                col_offset=33,
                end_lineno=16,
                end_col_offset=45)],
            [],
            lineno=16,
            col_offset=16,
            end_lineno=16,
            end_col_offset=46),
          lineno=16,
          col_offset=4,
          end_lineno=16,
          end_col_offset=46),
        Assign(
          [
            Name(
              'placeholder_11',
              Store(),
              lineno=17,
              col_offset=4,
              end_lineno=17,
              end_col_offset=14)],
          Call(
            Name(
              'placeholder_0',
              Load(),
              lineno=17,
              col_offset=17,
              end_lineno=17,
              end_col_offset=33),
            [
              Constant(
                5200418592.0,
                lineno=17,
                col_offset=34,
                end_lineno=17,
                end_col_offset=46)],
            [],
            lineno=17,
            col_offset=17,
            end_lineno=17,
            end_col_offset=47),
          lineno=17,
          col_offset=4,
          end_lineno=17,
          end_col_offset=47),
        FunctionDef(
          'placeholder_12',
          arguments(
            [],
            [
              arg(
                'self',
                lineno=19,
                col_offset=17,
                end_lineno=19,
                end_col_offset=21),
              arg(
                'placeholder_13',
                lineno=19,
                col_offset=23,
                end_lineno=19,
                end_col_offset=30)],
            kwonlyargs=[],
            kw_defaults=[],
            defaults=[]),
          [
            Assign(
              [
                Attribute(
                  Name(
                    'self',
                    Load(),
                    lineno=20,
                    col_offset=8,
                    end_lineno=20,
                    end_col_offset=12),
                  'placeholder_13',
                  Store(),
                  lineno=20,
                  col_offset=8,
                  end_lineno=20,
                  end_col_offset=20)],
              Name(
                'placeholder_13',
                Load(),
                lineno=20,
                col_offset=23,
                end_lineno=20,
                end_col_offset=30),
              lineno=20,
              col_offset=8,
              end_lineno=20,
              end_col_offset=30)],
          [],
          lineno=19,
          col_offset=4,
          end_lineno=20,
          end_col_offset=30)],
      [],
      lineno=8,
      col_offset=0,
      end_lineno=20,
      end_col_offset=30)],
  [])
## END NORMALIZED TREE ##